src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/PkcsPalOpenSsl.Encrypt.cs

```diff
@@ -37,26 +44,34 @@ internal sealed partial class PkcsPalOpenSsl : PkcsPal
 
                     ClassifyAndAddRecipients(cms, recipients);
 
-                    foreach (X509Certificate2 cert in originatorCerts)
+                    if (contentInfo.ContentType.Value != Oids.Pkcs7Data)
                     {
-                        using (SafeX509Handle certHandle = Interop.Crypto.X509Duplicate(cert.Handle))
+                        using (SafeAsn1ObjectHandle oid = Interop.Crypto.ObjTxt2Obj(contentInfo.ContentType.Value))
                         {
-                            Interop.Crypto.CheckValidOpenSslHandle(certHandle);
-                            status = Interop.Crypto.CmsAddOriginatorCert(cms, certHandle);
+                            Interop.Crypto.CheckValidOpenSslHandle(oid);
+                            status = Interop.Crypto.CmsSetEmbeddedContentType(cms, oid);
                             CheckStatus(status);
                         }
                     }
-                        
+
                     Interop.Crypto.BioWrite(contentBio, contentInfo.Content, contentInfo.Content.Length);
-                    status = Interop.Crypto.CmsCompleteMessage(cms, contentBio, false);
+
+                    // We have to do this to simulate the behavior for zero length content  
+                    bool detached = (contentInfo.Content.Length == 0);
+
+                    status = Interop.Crypto.CmsCompleteMessage(cms, contentBio, detached);
                     CheckStatus(status);
 
                     byte[] encryptedMessage = Interop.Crypto.OpenSslEncode(
                         handle => Interop.Crypto.CmsGetDerSize(handle),
                         (handle, buf) => Interop.Crypto.CmsEncode(handle, buf),
                         cms);
-
-                    // TODO(3334): Add support for unprotected attributes here.
+                    
+                    if (unprotectedAttributes.Count != 0)
+                    {
+                        // TODO(3334): Add support for unprotected attributes here.
+                        throw new NotImplementedException();
```

Comment:
On the one hand I'm not sure this really belongs with this change (it isn't fixing any bug or helping any test).  On the other, it is better than silently ignoring it in the event that we fail to come back to this.